### ISSUE TITLE
Make PackedCoordinateSequence Serializable

### ISSUE THREAD

**Issue by jandam** [locationtech/jts#21]
Please make PackedCoordinateSequence serializable. Other implementations of CoordinateSequence are Serializable.

Thank you

**Comment by rotulet** [comment]
I made a pull request #30. Wait and see... ;)

**Comment by jandam** [comment]
Hi, thank you for pull request.  I have another improvement for PackedCoordinateSequence.

There is Double precision for PackedCooridnateSequence. It is possible to create SoftReference in constructor    public Double(Coordinate[] coordinates, int dimension);

Add following line:
            this.coordRef = new SoftReference(coordinates);
This can improve performance and memory consumption.

**Comment by rotulet** [comment]
I let the contributors to check that point!

**Comment by dr-jts** [comment]
For code see https://github.com/rotulet/jts/commit/48f79636e0e72cafebc3684b5ae4115b63effc96#diff-1ef8c129b6c902d2df18f600afa3296e

The referenced PR got confused, so a new PR is needed.

**Comment by dr-jts** [comment]
Need a unit test as well.

**Comment by jandam** [comment]
Referenced code throws exception because field coordRef of type SoftReference is not Serializable.

```
public abstract class PackedCoordinateSequence
    implements CoordinateSequence, Serializable
{
  private static final long serialVersionUID = -3151899011275603L;

  /**
   * A soft reference to the Coordinate[] representation of this sequence.
   * Makes repeated coordinate array accesses more efficient.
   */
  protected transient SoftReference coordRef;

    /**
     * Builds a new packed coordinate sequence out of a coordinate array
     *
     * @param coordinates
     */
    public Double(Coordinate[] coordinates, int dimension) {
      if (coordinates == null)
        coordinates = new Coordinate[0];
      this.dimension = dimension;
      this.coordRef = new SoftReference<>(coordinates);

      coords = new double[coordinates.length * this.dimension];
      for (int i = 0; i < coordinates.length; i++) {
        coords[i * this.dimension] = coordinates[i].x;
        if (this.dimension >= 2)
          coords[i * this.dimension + 1] = coordinates[i].y;
        if (this.dimension >= 3)
          coords[i * this.dimension + 2] = coordinates[i].z;
      }
    }

    protected Object readResolve() throws ObjectStreamException {
        coordRef = null;
        return this;
    }

```

**Comment by dr-jts** [comment]
@jandam Thanks!

**Comment by dr-jts** [comment]
Fixed by #191
